rpc/rpc-lib/src/rpc-clnt.c

```diff
@@ -118,7 +120,7 @@ call_bail(void *data)
     pthread_mutex_unlock(&conn->lock);
     /*rpc_clnt_connection_cleanup will be unwinding all saved frames,
      * bailed or otherwise*/
-    if (!trans)
+    if (!trans || ctx->cleanup_started)
```

Comment:
What prevents `cleanup_started` from being set just after checking it here ? if so, the behavior would be the same and we would still have the timeout.

This reduces the race window, but doesn't eliminate the race.